cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Query.hs

```diff
@@ -78,101 +80,170 @@ hprop_cli_queries = integrationWorkspace "cli-queries" $ \tempAbsBasePath' -> H.
+        H.noteM_ $ execCli' execConfig [ eraName, "query", "protocol-parameters"
+                                       , "--out-file", protocolParametersOutFile ]
+        H.diffFileVsGoldenFile
+          protocolParametersOutFile
+          "test/cardano-testnet-test/files/golden/queries/protocolParametersFileOut.json"
+
+    TestQueryConstitutionHashCmd -> do
+      -- constitution-hash
+      -- Currently disabled (not accessible from the command line)
+      pure ()
+
+    TestQueryTipCmd ->
+      -- tip
+      do
+        -- to stdout
+        _ :: QueryTipLocalStateOutput <- H.noteShowM $ execCliStdoutToJson execConfig [ eraName, "query", "tip" ]
+        -- to a file
+        let tipOutFile = work </> "tip-out.json"
+        H.noteM_ $ execCli' execConfig [ eraName, "query", "tip"
+                                       , "--out-file", tipOutFile ]
+        _ :: QueryTipLocalStateOutput <- H.readJsonFileOk tipOutFile
+        pure ()
+
+    TestQueryStakePoolsCmd ->
+      -- stake-pools
+      do
+        -- to stdout
+        stakePoolsOut <- execCli' execConfig [ eraName, "query", "stake-pools" ]
+        H.assertWith stakePoolsOut $ \pools ->
+          length (lines pools) == 3 -- Because, by default, 3 stake pools are created
+        -- Light test of the query's answer, the ids should exist:
+        forM_ (lines stakePoolsOut) $ \stakePoolId -> do
+          execCli' execConfig [ eraName, "query", "pool-state"
+                              , "--stake-pool-id", stakePoolId ]
+        -- to a file
+        let stakePoolsOutFile = work </> "stake-pools-out.json"
+        H.noteM_ $ execCli' execConfig [ eraName, "query", "stake-pools" , "--out-file", stakePoolsOutFile]
+
+    TestQueryStakeDistributionCmd ->
+      -- stake-distribution
+      do
+        -- to stdout
+        stakeDistrOut <- execCli' execConfig [ eraName, "query", "stake-distribution" ]
+        -- stake addresses with stake
+        let stakeAddresses :: [(Text, Text)] =
+              map
+              ( bimap T.strip T.strip
+                . T.breakOn " " -- separate address and stake
+                . T.strip
+                . fromString )
+              . drop 2 -- drop header
+              . lines
+              $ stakeDistrOut
+        H.assertWith stakeAddresses $ \sa ->
+          -- Because, by default, 3 stake pools are created
+          length sa == 3
+        -- Light test of the query's answer, the ids should exist:
+        forM_ stakeAddresses $ \(stakePoolId, _) -> do
+          execCli' execConfig [ eraName, "query", "pool-state"
+                              , "--stake-pool-id", T.unpack stakePoolId ]
+        -- to a file
+        let stakePoolsOutFile = work </> "stake-distribution-out.json"
+        H.noteM_ $ execCli' execConfig [ eraName, "query", "stake-distribution"
+                                       , "--out-file", stakePoolsOutFile ]
+
+    TestQueryStakeAddressInfoCmd -> do
```

Comment:
We should test all the commands. This can be done in a follow up PR.

Reply:
That is just the first commit. The PR does implement tests for all commands